src/ir_MWM.cpp

```diff
@@ -0,0 +1,207 @@
+//
+// Status: Unimplemented.
+//
+void IRsend::sendMWM(uint8_t data[], uint16_t nbytes, uint16_t repeat) {
+  if (nbytes < 3)
+    return;  // Shortest possible message is 3 bytes
+
+  // Set 38kHz IR carrier frequency & a 1/4 (25%) duty cycle.
+  // NOTE: duty cycle is not confirmed. Just guessing based on RC5/6 protocols.
+  enableIROut(38, 25);
+
+  for (uint16_t r = 0; r <= repeat; r++) {
+    // Data
+    for (uint16_t i = 0; i < nbytes; i++) {
+      uint8_t byte = data[i];
+
+      // Start bit
+      mark(kMWMTick);
+
+      // LSB first, space=1
+      for (uint8_t mask = 0x1; mask; mask <<= 1) {
+        if (byte & mask) {  // 1
+          space(kMWMTick);
+        } else {  // 0
+          mark(kMWMTick);
+        }
+      }
+      // Stop bit
+      space(kMWMTick);
+    }
+    // Footer
+    space(kMWMMinGap);
+  }
+}
+#endif  // SEND_MWM
+
+#if DECODE_MWM
+// Decode the supplied MWM message.
+// This protocol is 2400 bps serial, 1 start bit (mark), 1 stop bit (space), no
+// parity
+//
+// Args:
+//   results: Ptr to the data to decode and where to store the decode result.
+//   nbits:   The number of data bits to expect.
+//   strict:  Flag indicating if we should perform strict matching.
+// Returns:
+//   boolean: True if it can decode it, false if it can't.
+//
+// Status: Unknown.
+//
+// Ref:
+//   http://www.sbprojects.com/knowledge/ir/rc5.php
+//   https://en.wikipedia.org/wiki/RC-5
+bool IRrecv::decodeMWM(decode_results *results, uint16_t nbits,
+    bool strict) {
+  DPRINTLN("DEBUG: decodeMWM");
+
+  // Compliance
+  if (results->rawlen < kMWMMinSamples) {
+    DPRINTLN("DEBUG: decodeMWM: too few samples");
+    return false;
+  }
+
+  uint16_t offset = kStartOffset;
+  uint16_t used = 0;
+  uint64_t data = 0;
+  uint16_t frame_bits = 0;
+  uint16_t data_bits = 0;
+
+  // No Header
+
+  // Data
+  uint8_t bits_per_frame = 10;
+  for (; offset < results->rawlen && results->bits < 8*kStateSizeMax; frame_bits++) {
+    DPRINT("DEBUG: decodeMWM: offset = ");
+    DPRINTLN(uint64ToString(offset));
+    int16_t level = getRClevel(results, &offset, &used, kMWMTick,
+        kMWMTolerance, kMWMExcess,
+        kMWMDelta, kMWMMaxWidth);
+    if (level < 0) {
+      DPRINTLN("DEBUG: decodeMWM: getRClevel returned error");
+      break;
+    }
+    switch (frame_bits % bits_per_frame) {
+      case 0:
+        // Start bit
+        if (level != kMark) {
+          DPRINTLN("DEBUG: decodeMWM: framing error - invalid start bit");
+          goto done;
+        }
+        break;
+      case 9:
+        // Stop bit
+        if (level != kSpace) {
+          DPRINTLN("DEBUG: decodeMWM: framing error - invalid stop bit");
+          return false;
+        }
+        else {
+          DPRINT("DEBUG: decodeMWM: data_bits = ");
+          DPRINTLN(data_bits);
+          DPRINT("DEBUG: decodeMWM: Finished byte: ");
+          DPRINTLN(data);
+          results->state[data_bits/8-1] = data & 0xFF;
```

Comment:
Just a style nitpick.
```suggestion
          results->state[data_bits / 8 - 1] = data & 0xFF;
```